src/EPR.Accreditation.Facade/Services/OverseasSiteService.cs

```diff
@@ -1,32 +1,53 @@
 ﻿using EPR.Accreditation.Facade.Common.Dtos;
 using EPR.Accreditation.Facade.Services.Interfaces;
 using EPR.Accreditation.Facade.Common.RESTservices.Interfaces;
+using EPR.Accreditation.Facade.Common.Dtos.Portal;
 
 namespace EPR.Accreditation.Facade.Services
 {
     public class OverseasSiteService : IOverseasSiteService
     {
         protected readonly IHttpOverseasSiteService _httpOverseasSiteService;
+        protected readonly IHttpCountryService _httpCountryService;
 
-        public OverseasSiteService(IHttpOverseasSiteService httpOverseasSiteService)
+        public OverseasSiteService(
+            IHttpOverseasSiteService httpOverseasSiteService,
+            IHttpCountryService httpCountryService)
         {
             _httpOverseasSiteService = httpOverseasSiteService ?? throw new ArgumentNullException(nameof(httpOverseasSiteService));
+            _httpCountryService = httpCountryService ?? throw new ArgumentNullException(nameof(httpCountryService));
         }
 
-        public async Task<OverseasAddress> GetReprocessorDetails(
+        public async Task<ReprocessorDetailsDto> GetReprocessorDetails(
             Guid accreditationExternalId,
             Guid overseasSiteExternalId)
         {
-            var overseasSite = await _httpOverseasSiteService.GetOverseasReprocessingSite(
+            var overseasSiteTask = _httpOverseasSiteService.GetOverseasReprocessingSite(
                 accreditationExternalId,
                 overseasSiteExternalId);
 
-            if (overseasSite.OverseasAddress == null)
+            var countriesTask = _httpCountryService.GetCountryList();
+
+            await Task.WhenAll(overseasSiteTask, countriesTask);
+
+            var overseasSite = overseasSiteTask.Result;
+            var countries = countriesTask.Result;
+
+            if (overseasSite.OverseasAddress == null
+                || !countries.Any())
             {
                 return null;
             }
 
-            return overseasSite.OverseasAddress;
+            var reprocessorDetails = new ReprocessorDetailsDto
```

Comment:
You could use AutoMapper here and populate the countries list later.... or use mapper to do countries as well